libraries/AP_Scripting/examples/battery_internal_resistance_check.lua

```diff
@@ -0,0 +1,35 @@
+--[[
+ script implementing pre-arm check that internal resistance is sensible
+--]]
+
+local MAX_RESISTANCE = 0.03
+
+local auth_id = arming:get_aux_auth_id()
```

Comment:
You could assert this to pass ci

Reply:
Could you explain this in shorter words, please?

I *am* struggling with the cleanliness checks...

Reply:
This is the error:
```
scripts/battery_internal_resistance_check.lua:30:34
        Code: param-type-mismatch
        Cannot assign `integer|nil` to parameter `integer`.
        - `nil` cannot match `integer`
        - Type `nil` cannot match `integer`
scripts/battery_internal_resistance_check.lua:24:37
        Code: param-type-mismatch
        Cannot assign `integer|nil` to parameter `integer`.
        - `nil` cannot match `integer`
        - Type `nil` cannot match `integer`
```

The errors are for:
```
arming:set_aux_auth_passed(auth_id)
```
and
```
arming:set_aux_auth_failed(auth_id, msg)
```

The issue is that both functions are expecting `auth_id` to be a integer type. However it is declared as:
```
local auth_id = arming:get_aux_auth_id()
```

The problem is that `arming:get_aux_auth_id()` can fail and return nil if its run out  of ids to give out. So `auth_id` could be a integer or it could be nil. 

If we change the definition of `auth_id` to:
```
local auth_id = assert(arming:get_aux_auth_id(), "No auth IDs available")
```
The assert will error out if the returned `auth_id` is nil, so now the checker knows that `auth_id` must be and integer. The types then match and CI is happy.

Reply:
> The assert will error out if the returned `auth_id` is nil, so now the checker knows that `auth_id` must be a integer after the assert. The types then match and CI is happy.

Many thanks, I now understand a little more about LUA :-)

I've made the suggested change to this PR.